airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -387,37 +276,59 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
         """
         return self._paginator.next_page_token(response, self._records_from_last_response)
 
+    def _fetch_next_page(
```

Comment:
As `_fetch_next_page` and `_read_pages` are copy/pasted from `HttpStream`, do we want to add a comment here and in `HttpStream` to refer to one another in case of changes? My guess is that any capability we offer to the SimpleRetriever will probably be something we want to grant to non declarative language streams. 

This also makes me believe that there is a layer of abstraction missing somewhere. This is out of scope for this PR but we should probably form a vision as to how our code should be structured to avoid duplicating code